frontend/projects/upgrade/src/app/features/dashboard/home/components/experiment-schedule/experiment-schedule.component.ts

```diff
@@ -134,75 +140,91 @@ export class ExperimentScheduleComponent implements OnInit {
             };
-            if (this.experimentInfo.state == this.ExperimentState.ENROLLING || this.experimentInfo.state == this.ExperimentState.ENROLLMENT_COMPLETE) {
-              this.emitExperimentDialogEvent.emit({
-                type: eventType,
-                formData: scheduleData,
-                path: NewExperimentPaths.EXPERIMENT_SCHEDULE
-              });
-            }
-          }
-          const { endExperimentAutomatically, endCondition, dateOfExperimentEnd, userCount, groupCount, dateOfExperimentStart } = this.experimentScheduleForm.value;
-          if (endExperimentAutomatically) {
-            switch (endCondition) {
-              case EndExperimentCondition.END_ON_DATE:
-                scheduleData = {
-                  ...scheduleData,
-                  endOn: dateOfExperimentEnd.toISOString()
-                };
-                break;
-
-              case EndExperimentCondition.END_CRITERIA:
-                scheduleData = {
-                  ...scheduleData,
-                  enrollmentCompleteCondition: {
-                    userCount: userCount || 0,
-                    groupCount: groupCount || 0
-                  }
-                };
-                break;
-            }
-          }
-          if (dateOfExperimentStart) {
+            break;
+          case EndExperimentCondition.END_CRITERIA:
             scheduleData = {
               ...scheduleData,
-              startOn: dateOfExperimentStart,
-              state: EXPERIMENT_STATE.SCHEDULED
-            }
-          } else {
-            if (this.experimentInfo) {
-              const { state } = this.experimentInfo;
-              scheduleData = {
-                ...scheduleData,
-                startOn: null,
-                state: state === EXPERIMENT_STATE.SCHEDULED ? EXPERIMENT_STATE.INACTIVE : state
+              enrollmentCompleteCondition: {
+                userCount: userCount || 0,
+                groupCount: groupCount || 0
               }
-            }
+            };
+            break;
+        }
+      }
+      if (dateOfExperimentStart) {
+        scheduleData = {
+          ...scheduleData,
+          startOn: dateOfExperimentStart,
+          state: EXPERIMENT_STATE.SCHEDULED
+        }
+      } else {
+        if (this.experimentInfo) {
+          const { state } = this.experimentInfo;
+          scheduleData = {
+            ...scheduleData,
+            startOn: null,
+            state: state === EXPERIMENT_STATE.SCHEDULED ? EXPERIMENT_STATE.INACTIVE : state
           }
-          this.emitExperimentDialogEvent.emit({
-            type: eventType,
-            formData: scheduleData,
-            path: NewExperimentPaths.EXPERIMENT_SCHEDULE
-          });
         }
-        break;
+      }
+      this.emitExperimentDialogEvent.emit({
+        type: eventType,
+        formData: scheduleData,
+        path: NewExperimentPaths.EXPERIMENT_SCHEDULE
+      });
     }
```

Comment:
i think we're missing a `break` statement

Reply:
saveData() function is called in switch case and there are break statement after function called there after flag reset.

Reply:
ok, that could be confusing hidden in a function, the `break` statement should be with the `case` code block to keep that syntax clean and readable

Reply:
> I've been thinking about this. What you have will work, but managing updates about data changes in this way in Angular is going to be painful in the long-run, because it's a lot of code to keep setting and unsetting flags and many components to pass data directly to each other. We have `ngrx` for this sort of thing.
> 
> Since this isn't a critical update, I have a request on this. Have you looked at `ngrx` and how it works in our app? This is a pretty perfect type of feature enhancement to learn how to use `ngrx` if not. Can you potentially try using `ngrx` actions and the `ngrx` data store to manage a boolean that indicates "dirty" or "clean" form state? We should be able to manage this feature without using `@Input` and `@Output`.

Let me know if this request makes sense, I'd be happy to show what I mean by this.

Reply:
![image](https://user-images.githubusercontent.com/50392803/199048122-4bd2bca6-7ecb-4a1b-a96f-2ffefc3325e3.png)
They are in `case` code block, I just put code below `SAVE_DATA` in function and called it for both `SEND_FORM_DATA` and `SAVE_DATA` cases.

Reply:
ok apologies, reading it in github-diff made me see it weird then